src/main/scala/molmed/qscripts/DNABestPracticeVariantCalling.scala

```diff
@@ -315,31 +323,37 @@ class DNABestPracticeVariantCalling extends QScript
 
       val splitsBams = runChromosomeSplitting(bams, groupsToSplitTo, generalUtils, reference)
 
-      val splitAndProcessedBams =
+      val splitAndProcessedBamTargets =
         for (splitGroup <- splitsBams) yield {
-          val processedBamFiles = gatkDataProcessingUtils.dataProcessing(
+          val processedBamTargets = gatkDataProcessingUtils.dataProcessing(
             splitGroup, processedAligmentsOutputDir, cleaningModel,
             skipDeduplication = false, testMode)
-          processedBamFiles
+          processedBamTargets
         }
 
-      for (toMergeBams <- splitAndProcessedBams) yield {
+      for (toMergeBamTarget <- splitAndProcessedBamTargets) yield {
 
         // Assumes that the start of the file name is the same, and is what is to
         // be used name these files.
-        val nameOfOutputBam =
-          if (toMergeBams.size > 1) {
-            val firstFileName = toMergeBams(0).getName()
-            val secondFileName = toMergeBams(1).getName()
-            val longestCommonName =
-              firstFileName.zip(secondFileName).takeWhile(Function.tupled(_ == _)).map(_._1).mkString
-            // The splitting will add a _, removing it here.  
-            longestCommonName.stripSuffix("_")
-          } else
-            toMergeBams(0).getName().stripSuffix(".bam")
-
-        val outBam = new File(processedAligmentsOutputDir + "/" + nameOfOutputBam + ".bam")
-        SplitFilesAndMergeByChromosome.merge(qscript, toMergeBams, outBam, asIntermediate = false, generalUtils)
+        
+        def _longestCommonPrefix(fileNames: Seq[String]): String = {
+          if (fileNames.size > 1)
+              fileNames(0).zip(fileNames(1)).takeWhile(Function.tupled(_ == _)).map(_._1).mkString.stripSuffix("_")
```

Comment:
Maybe break up this to multiple lines for read-ability? I realize that was not the case in the original code but I do think it would be better that way.